Frends.PDF.Create/Frends.PDF.Create/Create.cs

```diff
@@ -0,0 +1,430 @@
+            [PropertyTab] DocumentSettings documentSettings,
+            [PropertyTab] DocumentContent content,
+            [PropertyTab] Options options
+        )
+        {
+            try
+            {
+                var document = new Document();
+                if (!string.IsNullOrWhiteSpace(documentSettings.Title)) document.Info.Title = documentSettings.Title;
+                if (!string.IsNullOrWhiteSpace(documentSettings.Author)) document.Info.Author = documentSettings.Author;
+
+                // Get the selected page size.
+                PageSetup.GetPageSize(documentSettings.Size.ConvertEnum<PageFormat>(), out Unit width, out Unit height);
+                var section = document.AddSection();
+                SetupPage(section.PageSetup, width, height, documentSettings);
+
+                // Index for stylename.
+                var elementNumber = 0;
+
+                // Add page elements.
+                foreach (var pageElement in content.Contents)
+                {
+                    var styleName = $"style_{elementNumber}";
+                    var style = document.Styles.AddStyle(styleName, "Normal");
+
+                    switch (pageElement.ContentType)
+                    {
+                        case ElementType.Image:
+                            AddImage(section, pageElement, width);
+                            break;
+                        case ElementType.PageBreak:
+                            section = document.AddSection();
+                            SetupPage(section.PageSetup, width, height, documentSettings);
+                            break;
+                        case ElementType.Header:
+                            SetFont(style, pageElement);
+                            SetParagraphStyle(style, pageElement);
+                            AddHeaderFooterContent(section, pageElement, style, true);
+                            break;
+                        case ElementType.Footer:
+                            SetFont(style, pageElement);
+                            SetParagraphStyle(style, pageElement);
+                            AddHeaderFooterContent(section, pageElement, style, false);
+                            break;
+                        case ElementType.Table:
+                            AddTable(section, pageElement, width);
+                            break;
+                        default:
+                            SetFont(style, pageElement);
+                            SetParagraphStyle(style, pageElement);
+                            AddTextContent(section, pageElement, style);
+                            break;
+                    }
+
+                    elementNumber++;
+                }
+
+                var fileName = Path.Combine(outputFile.Directory, outputFile.FileName);
+                var fileNameIndex = 1;
+
+                while (File.Exists(fileName) && outputFile.FileExistsAction != FileExistsActionEnum.Overwrite)

```

Comment:
This method is huge and could benefit from some splitting - e.g. getting file name can definitely be one separate helper, adding sections could be separate method etc. Maybe just a bit better code organisation?

Reply:
Moved task method functionality to helper methods.